os/mm/mm_heap/mm_getsize.c

```diff
@@ -0,0 +1,90 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include <sys/types.h>
+#include <tinyara/mm/mm.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static size_t mm_find_largest_freesize(struct mm_heap_s *heap)
+{
+	size_t largest_size = 0;
+	struct mm_freenode_s *fnode;
+	int i = 0;
+
+	/* Free nodes are sorted in a descending order,
+	 * so the first node in each nodelist is the largest within its nodelist.
+	 */
+	for (i = 0; i < MM_NNODES; ++i) {
+		fnode = heap->mm_nodelist[i].flink;
+		if (fnode && largest_size < fnode->size) {
+			largest_size = fnode->size;
+		}
+	}
+	return largest_size;
+}
+
+/****************************************************************************
+ * Name: mm_get_largest_freenode_size
+ *
+ * Description:
+ *   returns a largest node size in the heap
+ ****************************************************************************/
+
+size_t mm_get_largest_freenode_size(void *address)
+{
+	struct mm_heap_s *heap;
+	size_t largest_size = 0;
+
+#ifdef __KERNEL__
+	size_t cheap_fsize;
```

Comment:
The name of this variable is not clear. Why do we call it cheap_fsize? Can we change this to something easily understandable?

Reply:
Changed to curheap_freesize.
Thanks.